### ISSUE TITLE
Improve compatibility logic for the EventDispatcher

### ISSUE THREAD

**Issue by TavoNiievez** [Codeception/phpunit-wrapper#90]
See failed job in  https://github.com/Codeception/module-laravel/pull/12 . Caused by:

https://github.com/Codeception/phpunit-wrapper/blob/f5d311f4df711f7f37475e5a9190c2c4aef5518a/src/DispatcherWrapper.php#L20-L26

Suggested change (based on https://github.com/lexik/LexikJWTAuthenticationBundle/pull/669/files):

```diff
- use Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcherInterface;
+ use Symfony\Contracts\EventDispatcher\EventDispatcherInterface as ContractsEventDispatcherInterface;

//...

-   if (!interface_exists(TraceableEventDispatcherInterface::class)) {
+   if ($dispatcher instanceof ContractsEventDispatcherInterface) {
        $dispatcher->dispatch($eventObject, $eventType);
    } else {
        $dispatcher->dispatch($eventType, $eventObject);
    }
```

Related to: https://github.com/Codeception/Codeception/issues/6065

**Comment by Naktibalda** [comment]
It's complicated. I changed this logic at least once and it still has caveats.
Changing it again can break tests for someone else.

It is on the list of things I'm going to remove in Codeception 5.
I think that it would be the best to wait for Codeception 5 now.

**Comment by Naktibalda** [comment]
Alright, I looked at Symfony code and this proposal looks good, please raise pull requests to 6.0, 6.5, 7.1, 8.0 and 9.0 branches.

**Comment by TavoNiievez** [comment]
I already sent the PRs for all the branches of this repo. May I send one also for the `4.1` branch of `Codeception\Codeception`?

**Comment by Naktibalda** [comment]
Yes, please update 4.1 branch of codeception/codeception too.

**Comment by TavoNiievez** [comment]
Thanks, see https://github.com/Codeception/Codeception/pull/6066 , this issue can be closed now.

**Comment by Naktibalda** [comment]
I released new versions of all branches of phpunit-wrapper.

**Comment by TavoNiievez** [comment]
thanks!

Do I require `4.1.x-dev` of `Codeception/Codeception` in Laravel module **CI job** or will you create a `4.1.14` tag?

**Comment by Naktibalda** [comment]
I released 4.1.14
